docs/howto/build-guides/platforms.rst

```diff
@@ -0,0 +1,169 @@
+produce one charm that runs on an ``amd64`` Ubuntu 24.04 system.
+
+.. note::
+
+  The OS of the build system doesn't affect what is built. Using LXD or
+  Multipass as a backend, Charmcraft will create a build environment with the
+  correct OS.
+
+Create a charm for all architectures
+------------------------------------
+
+For charms that are architecture-independent, use the following
+``charmcraft.yaml`` snippet to create a single charm that can run on any
+architecture:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     all:
+       build-on: [amd64]
+       build-for: [all]
+
+Building on ``amd64`` will produce one charm that runs on any architecture on
+an Ubuntu 24.04 system.
+
+The ``all`` keyword can only be used for the ``build-for`` keyword. If the
+charm can be built on multiple architectures, they must be added to the
+``build-on`` keyword:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     all:
+       build-on: [amd64, riscv64]
+       build-for: [all]
+
+Building on ``amd64`` or ``riscv64`` will produce one charm that runs on all
+architectures on Ubuntu 24.04 systems.
+
+
+Create a set of charms for multiple architectures
+-------------------------------------------------
+
+Charms may contain architecture-specific code. To build charms for specific
+architectures, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+       build-on: [amd64]
+       build-for: [amd64]
+     riscv64:
+       build-on: [riscv64]
+       build-for: [riscv64]
+
+Because the ``build-on`` and ``build-for`` entries are identical for each
+platform, the shorthand notation can be used instead:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+     riscv64:
+
+The results are the same with either snippet. Building on ``amd64`` will
+produce one charm that runs on ``amd64``. Building on ``riscv64`` will produce
+one charm that runs on ``riscv64``.
+
+
+Create a charm for a different architecture
+-------------------------------------------
+
+A charm may require cross-compilation to build. To create a charm for a
+different architecture, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     riscv64-cross:
+       build-on: [amd64]
+       build-for: [riscv64]
+
+Building on ``amd64`` will produce one charm that runs on ``riscv64``.
+
+Note that the charm developer must ensure the charm is compatible with the
+target architectures. By default, the `charm`_,
+:ref:`python <craft_parts_python_plugin>`, :ref:`poetry <craft_parts_poetry_plugin>`,
+and :ref:`uv <craft_parts_uv_plugin>` plugins will install wheels for python packages
+for the ``build-on`` architecture rather than the ``build-for``. For more information,
+see `craft-parts#974`_.
+
+Create a set of charms for multiple OS releases
```

Comment:
```suggestion
Select multiple platform bases
```

Alternatively, "multiple bases".

Reply:
I'd rather not use the term "bases" if possible since it could confuse people about the deprecated `bases` keyword that `base` and `platforms` replace. For now I've gone with "Select multiple OS releases", but we should discuss this further.